File: Pôle Liaison & Protocole/test_protocol.c
#include <stdio.h>
#include <string.h>
#include "protocol.h"
// On part du principe que tu as mis le code du TCPBuffer et 
// les structures Message / ActionType juste au-dessus ici.

int main() {
    printf("--- DEBUT DU CRASH-TEST C ---\n");

    // 1. Initialisation de notre buffer
    TCPBuffer tcp_buf;
    init_buffer(&tcp_buf);
    char msg_extrait[100]; // Pour stocker le résultat

    // 2. SIMULATION DE COUPURE
    printf("\n[Réseau] On reçoit un bout de message...\n");
    add_data(&tcp_buf, "1|120|250|", 10); // Il manque la fin !

    if (get_next_message(&tcp_buf, msg_extrait)) {
        printf("-> Erreur: Il a trouvé un message alors qu'il n'y a pas de '\\n' !\n");
    } else {
        printf("-> Correct: Aucun message complet trouve.\n");
    }

    // 3. SIMULATION DE SPAM (La fin arrive, avec deux autres messages)
    printf("\n[Réseau] On reçoit la suite, avec des messages collés...\n");
    add_data(&tcp_buf, "0|None\n2|50|50|1|Pb1\n3|300|300|2|Pc", 35);
    add_data(&tcp_buf, "3\n", 2); // On rajoute le petit bout qui manquait au 3eme

    printf("\n--- EXTRACTION DES MESSAGES ---\n");
    // On boucle tant qu'on trouve des '\n'
    while (get_next_message(&tcp_buf, msg_extrait)) {
        printf("\nMessage brut extrait : %s\n", msg_extrait);

        // 4. On teste la désérialisation (sscanf)
        Message msg_dest;
        if (deserialize_message(msg_extrait, &msg_dest)) {
            printf("Succès ! Objet C recréé : Joueur %d en (%d, %d)\n", 
                   msg_dest.id_joueur, msg_dest.pos_x, msg_dest.pos_y);
        } else {
            printf("Erreur de désérialisation (sscanf a échoué).\n");
        }
    }

    return 0;
}

File: Pôle Liaison & Protocole/test_protocol.py
# Fichier : test_protocol.py
from protocol import Message, ActionType
from tampon import TCPBuffer

print("--- DEBUT DU CRASH-TEST PYTHON ---")

# 1. On crée un faux buffer réseau
buffer = TCPBuffer()

# 2. SIMULATION DE COUPURE : Le réseau est lent
print("\n[Réseau] On reçoit un demi-message...")
buffer.add_data(b"1|120|250|0|No") # Attention, pas de \n à la fin !

message_lu = buffer.get_next_message()
print(f"-> Résultat de la lecture : {message_lu} (C'est normal, il manque la fin !)")

# 3. SIMULATION DE SPAM : La fin du message arrive, collée à 2 autres messages
print("\n[Réseau] On reçoit la fin, suivie de deux autres messages collés...")
# "ne\n" finit le premier message. Puis on colle le joueur 2 et le joueur 3.
buffer.add_data(b"ne\n2|50|50|1|Pb1\n3|300|300|2|Pc3\n")

print("\n--- EXTRACTION DES MESSAGES ---")
while True:
    msg_str = buffer.get_next_message()
    if msg_str is None:
        print("-> Plus de messages complets dans le buffer.")
        break
    
    print(f"\nMessage brut lu : {msg_str}")
    
    # 4. On vérifie que la désérialisation fonctionne
    try:
        msg_obj = Message.deserialize(msg_str)
        print(f"Succès ! Objet recréé : Le joueur {msg_obj.id_joueur} fait l'action n°{msg_obj.action.value} vers '{msg_obj.target_id}'")
    except Exception as e:
        print(f"Erreur de lecture du message : {e}")